File: Models/Partitura.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using Maia.Models.Abstracts;
using Maia.Models.DTO;
using Newtonsoft.Json;

namespace Maia.Models
{
    public class Partitura : AnotacaoMusical
    {
        [MaxLength(90)]
        public string Titulo { get; set; }
        
        [Required]
        public string UrlPdf { get; set; }
        
        public Partitura() { }

        public Partitura(PartituraDTO dto)
        {
            this.Titulo = dto.Titulo;
            this.UrlPdf = dto.UrlPdf;
            this.MusicaId = dto.MusicaId;
            this.UsuarioId = dto.UsuarioId;
        }
    }
}

File: Angular/src/app/shared/models/Partitura.ts
import {AnotacaoMusical} from "./AnotacaoMusical";
import {Usuario} from "./Usuario";
import {Musica} from "./Musica";

export class Partitura extends AnotacaoMusical {
  public titulo: string;
  public urlPdf: string;

  constructor(titulo: string, urlPdf: string, musicaId: number, usuarioId: number, id?: number, usuario?: Usuario, musica?: Musica) {
    super(musicaId, usuarioId, usuario, musica, id);
    this.titulo = titulo;
    this.urlPdf = urlPdf;
  }
}
